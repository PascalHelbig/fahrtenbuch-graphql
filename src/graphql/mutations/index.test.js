const { graphql } = require('graphql');
const { makeExecutableSchema } = require('graphql-tools');
const Mutation = require('./index');

jest.mock('../../models/User.js');

it('should list all mutations', () => {
  const SchemaDefinition = `
    ${Mutation.schema}
    type Query { ignore: Boolean }
    schema { mutation: Mutation, query: Query } 
  `;
  const schema = makeExecutableSchema({ typeDefs: [SchemaDefinition, Mutation.usedTypes] });
  const query = `
    {
      __schema {
        mutationType {
          fields {
            name
            args { name }
            type { name }
          }
        }
      }
    }
  `;

  return graphql(schema, query).then(result =>
    expect(result).toMatchSnapshot()
  );
});
