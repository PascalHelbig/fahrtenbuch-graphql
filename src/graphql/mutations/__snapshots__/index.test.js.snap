exports[`test should list all mutations 1`] = `
Object {
  "data": Object {
    "__schema": Object {
      "mutationType": Object {
        "fields": Array [
          Object {
            "args": Array [
              Object {
                "name": "email",
              },
              Object {
                "name": "password",
              },
            ],
            "name": "login",
            "type": Object {
              "name": "Login",
            },
          },
          Object {
            "args": Array [
              Object {
                "name": "email",
              },
              Object {
                "name": "password",
              },
            ],
            "name": "signup",
            "type": Object {
              "name": "Login",
            },
          },
          Object {
            "args": Array [
              Object {
                "name": "token",
              },
              Object {
                "name": "group",
              },
            ],
            "name": "addGroup",
            "type": Object {
              "name": "Group",
            },
          },
          Object {
            "args": Array [
              Object {
                "name": "token",
              },
              Object {
                "name": "boat",
              },
            ],
            "name": "addUserBoat",
            "type": Object {
              "name": "Boat",
            },
          },
          Object {
            "args": Array [
              Object {
                "name": "token",
              },
              Object {
                "name": "entry",
              },
              Object {
                "name": "participations",
              },
            ],
            "name": "addEntry",
            "type": Object {
              "name": "Entry",
            },
          },
        ],
      },
    },
  },
}
`;
